extensions/applicationinsights-properties-js/src/Context/Session.ts

```diff
@@ -99,120 +120,125 @@ export class _SessionManager {
-                        "Error parsing ai_session cookie, session will be reset: " + Util.getExceptionName(e),
-                        { exception: dumpObj(e) });
-                }
-        
-                if (_self.automaticSession.renewalDate === 0) {
-                    _logger.throwInternal(LoggingSeverity.WARNING,
-                        _InternalMessageId.SessionRenewalDateIsZero,
-                        "AI session renewal date is 0, session will be reset.");
                 }
+
+                return isValid;
             }
         
-            function _renew() {
-                const now = new Date().getTime();
-        
-                _self.automaticSession.id = Util.newId((_self.config && _self.config.idLength) ? _self.config.idLength() : 22);
+            function _renew(now: number) {
+                let theConfig = (_self.config ||{});
+                let getNewId = (theConfig.getNewId ? theConfig.getNewId() : null) || newId;
+                _self.automaticSession.id = getNewId(theConfig.idLength ? theConfig.idLength() : 22);
                 _self.automaticSession.acquisitionDate = now;
-                _self.automaticSession.renewalDate = now;
         
-                _setCookie(_self.automaticSession.id, _self.automaticSession.acquisitionDate, _self.automaticSession.renewalDate);
+                _setCookie(_self.automaticSession, now);
         
                 // If this browser does not support local storage, fire an internal log to keep track of it at this point
-                if (!Util.canUseLocalStorage()) {
+                if (!utlCanUseLocalStorage()) {
                     _logger.throwInternal(LoggingSeverity.WARNING,
                         _InternalMessageId.BrowserDoesNotSupportLocalStorage,
                         "Browser does not support local storage. Session durations will be inaccurate.");
                 }
             }
         
-            function _setCookie(guid: string, acq: number, renewal: number) {
+            function _setCookie(session: ISession, now: number) {
+                let acq = session.acquisitionDate;
+                session.renewalDate = now
+
+                let config = _self.config;
+                let renewalPeriod = config.sessionRenewalMs();
+
                 // Set cookie to expire after the session expiry time passes or the session renewal deadline, whichever is sooner
                 // Expiring the cookie will cause the session to expire even if the user isn't on the page
-                const acquisitionExpiry = acq + _self.config.sessionExpirationMs();
-                const renewalExpiry = renewal + _self.config.sessionRenewalMs();
-                const cookieExpiry = new Date();
-                const cookie = [guid, acq, renewal];
+                const acqTimeLeft = (acq + config.sessionExpirationMs()) - now;
+                const cookie = [session.id, acq, now];
+                let maxAge = 0;
```

Comment:
nit: I see lot of variables based on time. In my opinion it would be easy to read if names have the suffix based on measurement of time. In this case 'maxAgeSecs' and 'acqTimeLeftMs'

Reply:
Very reasonable suggestion